### PR TITLE
feat: support tagging multiple tags at once for oras push

### PR DESCRIPTION
**Pull request by wju-MSFT**
Signed-off-by: Jun <wju@microsoft.com>

### CODE REVIEW

**Comment by wju-MSFT** [comment]
1) push new file with 3 tags.
Preparing /home/wju/demo/h.txt
Uploading 44136fa355b3 application/vnd.unknown.config.v1+json
Uploading 91ee5e9f42ba /home/wju/demo/h.txt
Uploaded  91ee5e9f42ba /home/wju/demo/h.txt
Uploaded  44136fa355b3 application/vnd.unknown.config.v1+json
Uploading 8cf27269e898 application/vnd.oci.image.manifest.v1+json
Uploaded  8cf27269e898 application/vnd.oci.image.manifest.v1+json
Pushed localhost:12900/hello:tag1
Tagged tag2
Tagged tag3
Digest: sha256:8cf27269e898510020eff45bd4a16985a194a5f8bb677e99bd0f6cf883937b11

2) push existed file with 3 tags
Preparing /home/wju/demo/h.txt
Exists    8cf27269e898 application/vnd.oci.image.manifest.v1+json
Pushed localhost:12900/hello:tag1
Tagged tag3
Tagged tag2
Digest: sha256:8cf27269e898510020eff45bd4a16985a194a5f8bb677e99bd0f6cf883937b11

File: cmd/oras/push.go [hunk 1]
```diff
@@ -39,14 +41,16 @@ type pushOptions struct {
...
 	manifestConfigRef string
 	artifactType      string
+	concurrency       int64
 }
 
 func pushCmd() *cobra.Command {
 	var opts pushOptions
 	cmd := &cobra.Command{
-		Use:   "push [flags] <name>[:<tag>|@<digest>] <file>[:<type>] [...]",
+		Use:   "push [flags] <name>[:<tag>[,<tag>] [...]] <file>[:<type>] [...]",
```
Comment (shizhMSFT) [inline cmd/oras/push.go:53]:
```suggestion
		Use:   "push [flags] <name>[:<tag>[,<tag>][...]] <file>[:<type>] [...]",
```

File: cmd/oras/push.go [hunk 2]
```diff
@@ -76,6 +80,12 @@ Example - Push repository with manifest annotations
 
 Example - Push repository with manifest annotation file
   oras push --annotation-file annotation.json localhost:5000/hello:latest
+
+Example - Push file "hi.txt" with multiple tags:
+  oras push localhost:5000/hello:tag1,tag2,tag3 hi.txt
+
+Example - Push file "hi.txt" with multiple tags and concurrency level tuned:
+	oras push --concurrency 6 localhost:5000/hello:tag1,tag2,tag3 hi.txt
```
Comment (shizhMSFT) [inline cmd/oras/push.go:88]: incorrect indentation

File: cmd/oras/push.go [hunk 3]
```diff
@@ -85,14 +95,17 @@ Example - Push repository with manifest annotation file
...
+			opts.targetRef = refs[0]
+			opts.extraRefs = refs[1:]
 			opts.FileRefs = args[1:]
 			return runPush(opts)
 		},
 	}
 
 	cmd.Flags().StringVarP(&opts.manifestConfigRef, "config", "", "", "manifest config file")
 	cmd.Flags().StringVarP(&opts.artifactType, "artifact-type", "", "", "media type of the manifest config")
+	cmd.Flags().Int64VarP(&opts.concurrency, "concurrency", "", 5, "concurrency level")
```
Comment (shizhMSFT) [inline cmd/oras/push.go:108]: Concurrency should also be effective in copy options.

File: cmd/oras/push.go [hunk 4]
```diff
@@ -145,6 +158,18 @@ func runPush(opts pushOptions) error {
 	}
 
 	fmt.Println("Pushed", opts.targetRef)
+
+	contentBytes, err := content.FetchAll(ctx, store, desc)
+	if err != nil {
+		return err
+	}
```
Comment (shizhMSFT) [inline cmd/oras/push.go:165]: We don't need to fetch if `len(opts.extraRefs)` is 0.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: cmd/oras/push.go (hunk 1 above)
Comment (wju-MSFT) [reply cmd/oras/push.go:53]: done

File: cmd/oras/push.go (hunk 2 above)
Comment (wju-MSFT) [reply cmd/oras/push.go:88]: done

File: cmd/oras/push.go (hunk 3 above)
Comment (wju-MSFT) [reply cmd/oras/push.go:108]: done

File: cmd/oras/push.go (hunk 4 above)
Comment (wju-MSFT) [reply cmd/oras/push.go:165]: good point!

File: cmd/oras/push.go [hunk 5]
```diff
@@ -76,6 +80,12 @@ Example - Push repository with manifest annotations
 
 Example - Push repository with manifest annotation file
   oras push --annotation-file annotation.json localhost:5000/hello:latest
+
+Example - Push file "hi.txt" with multiple tags:
+  oras push localhost:5000/hello:tag1,tag2,tag3 hi.txt
+
+Example - Push file "hi.txt" with multiple tags and concurrency level tuned:
+  oras push --concurrency 6 localhost:5000/hello:tag1,tag2,tag3 hi.txt
   `,
```
Comment (shizhMSFT) [inline cmd/oras/push.go:89]: nit:
```suggestion
`,
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by shizhMSFT** [comment]
This PR is related to #355

**Pull request merged**